tests/Service/GroupsIntegrationTest.php

```diff
@@ -146,6 +152,83 @@ public function testExpand2() {
 
 	}
 
+	public function testExpandUmlauts() {
+		$recipients = "john@doe.com,namespace1:ümlaut";
+		$members = [
+			[
+				'id' => 'bob',
+				'name' => "Bobby",
+				'email' => "bob@smith.net"
+			],
+			[
+				'id' => 'mary',
+				'name' => 'Mary',
+				'email' => 'mary@smith.net'
+			]
+		];
+		$this->groupService1->expects($this->once())
+			->method('getUsers')
+			->willReturn($members);
+
+		$expected = "john@doe.com,bob@smith.net,mary@smith.net";
+
+		$actual = $this->groupsIntegration->expand($recipients);
+
+		$this->assertEquals($expected, $actual);
+	}
+
+	public function testExpandSpace() {
+		$recipients = "john@doe.com,namespace1:test group";
+		$members = [
+			[
+				'id' => 'bob',
+				'name' => "Bobby",
+				'email' => "bob@smith.net"
+			],
+			[
+				'id' => 'mary',
+				'name' => 'Mary',
+				'email' => 'mary@smith.net'
+			]
+		];
+		$this->groupService1->expects($this->once())
+			->method('getUsers')
+			->with('test group')
+			->willReturn($members);
+
+		$expected = "john@doe.com,bob@smith.net,mary@smith.net";
+
+		$actual = $this->groupsIntegration->expand($recipients);
+
+		$this->assertEquals($expected, $actual);
+	}
+
+	public function testExpandLabeledGroup() {
+		$recipients = "\"test group (Nextcloud)\" <nextcloud:test group>";
```

Comment:
@myrho this is what I get from the groups integration, sent to the server as is. Was the server ever able to understand/parse this?

Reply:
I don't get it. What server?

Reply:
When the front-end does autocompletion, it gets an entry with
```json
{
  "label": "test group (Nextcloud)",
  "address": "nextcloud:test group"
}
```

If the user then send the email, the `to` HTTP parameters becomes `"test group (Nextcloud)" <nextcloud: test group>`.  This falls apart on the group expansion as the regex would detect `test group>` as group name.

Do you remember if we always did it this way? As in, sending the address **and** label? The group expansion would handle the address well, but the combination with label causes nothing but problems.

Reply:
Sorry for the confusion caused. What I was looking for is this code: https://github.com/nextcloud/mail/blob/c89121aa5d2b3560a102dd2cb259dd017b5692dc/src/components/Composer.vue#L313. It needs an update for the new group names we allow (umlauts, spaces and others) :)